File: scripts/check-translation-fields.js
#!/usr/bin/env node

/**
 * Check what fields exist in translations table
 */

const Airtable = require('airtable');
require('dotenv').config();

const base = new Airtable({ apiKey: process.env.AIRTABLE_TOKEN_CG })
  .base(process.env.AIRTABLE_BASE_ID_CG);

const TRANSLATIONS_TABLE = process.env.AIRTABLE_TRANSLATIONS_TABLE_ID_CG;

async function checkFields() {
  console.log('\n🔍 Checking Translation Table Fields\n');

  // Get first PT record to see structure
  const records = await base(TRANSLATIONS_TABLE)
    .select({
      filterByFormula: '{language} = "pt"',
      maxRecords: 1
    })
    .all();

  if (records.length === 0) {
    console.log('❌ No PT translations found');
    return;
  }

  const record = records[0];
  console.log('📋 Available fields in translation record:');
  console.log(Object.keys(record.fields));
  console.log('\n📝 Sample field values:');
  for (const [key, value] of Object.entries(record.fields)) {
    if (typeof value === 'string' && value.length > 100) {
      console.log(`   ${key}: [string, length: ${value.length}]`);
    } else if (Array.isArray(value)) {
      console.log(`   ${key}: [array, length: ${value.length}] - ${JSON.stringify(value)}`);
    } else {
      console.log(`   ${key}:`, value);
    }
  }
}

checkFields().catch(console.error);

File: check-translation-fields.py
#!/usr/bin/env python3
import os
import requests

AIRTABLE_TOKEN = os.getenv('AIRTABLE_TOKEN_CG')
AIRTABLE_BASE_ID = os.getenv('AIRTABLE_BASE_ID_CG')
API_URL = f'https://api.airtable.com/v0/{AIRTABLE_BASE_ID}/Companion_Translations'

headers = {'Authorization': f'Bearer {AIRTABLE_TOKEN}'}

response = requests.get(f'{API_URL}?maxRecords=1&filterByFormula={{language}}="nl"', headers=headers)
data = response.json()

if data.get('records'):
    fields = data['records'][0]['fields']
    print("Available fields in Companion_Translations (NL):")
    print("=" * 60)
    for field_name, value in sorted(fields.items()):
        value_preview = str(value)[:100] if value else "(empty)"
        print(f"  {field_name}: {value_preview}")
